docs/developers/content-creation/creating-a-role.md

```diff
@@ -40,3 +44,526 @@ suicideMultiplier = -1
+Open the linked GMOD Wiki page to understand the hook better. To be able to use a hook, we need to add it to our code. We can do this with the `hook.Add()` command. Garry's Mod expects us to supply 3 parameters with this command. The first place is for the name of the hook we want to call. In our case "PlayerDeath" . The second one is for the identifier. This is used to give the hook a name that can be used to recognize it. To prevent overwriting, this identifier should be a bit more extensive. It is probably best to name the hook `ttt2_heal_up_tester`. The third and last place is reserved for the function to be executed when the hook is called. This function in turn also has 3 parameters that we can use: `victim, inflictor, attacker`. As described in the GMOD Wiki article, "victim" is the player who died, "inflictor" is the item used for this and "attacker" is the player or entity that killed the "victim".
+
+```lua
+hook.Add("PlayerDeath", "ttt2_heal_up_tester", function(victim, inflictor, attacker)
+    -- Code to be processed must be filled in here
+end)
+```
+
+Now, before we program the actual function, we will program a few conditions that will terminate the function immediately, since these cases are not relevant to the role. 
+
+For this purpose we use so-called "if-statements". This can be used to create function blocks that are only called if the associated condition has been met. If you already have some experience, you can first try to define 2 of the 3 conditions yourself. The third condition is a bit more special. If you have problems or can't do that yet, you can scroll down further. The first condition is to prevent that the victim is also a tester. If the victim is a tester, the function should be terminated immediately. The other condition is to terminate the function if the victim is not a traitor. After all, we only want to heal the tester if the person who died was really a traitor colleague.
+
+```lua
+    if victim:GetSubRole() == ROLE_TESTER then return end
+  
+    if victim:GetTeam() ~= TEAM_TRAITOR then return end
+
+    if SpecDM and (victim.IsGhost() and victim:IsGhost()) then return end
+```
+
+Theoretically, you can put these 3 statements directly below each other. So that you can understand them better, I have inserted paragraphs between them. 
+Before we go any further, let's try to understand these statements now.
+
+`if victim:GetSubRole() == ROLE_TESTER then return end`
+
+“==” means something like "is equal". "~=" would mean "is unequal". Basically the line means "If the victim is tester as subrole" (if victim:GetSubRole() == ROLE_TESTER), then terminate the function (then return end). At this point it is important to access the subrole. If the base role were accessed, the tester would be a traitor.
```

Comment:
`“==” means something like "is equal". "~=" would mean "is unequal".` That's probably a bit too much. If you don't know what `==` means, you shouldn't code a role